### PR TITLE
SingleFileGenerator and ResX generation support

### PR DESCRIPTION
**Pull request by 333fred**
This adds support for creating pkgdef elements that specify SingleFileGenerators in remote assemblies. I've also defined the SFG's for ResX code file generation. However, the designer currently expects the `IVsHierarchy` to implement `IVsSingleFileGeneratorFactory`, and it does not, the combobox is disabled. Manually changing the CustomTool property, or using the Run Custom Tool entry in the right-click project menu, produces the right results.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/Packaging/CSharpProjectSystemPackage.cs [hunk 1]
```diff
@@ -13,6 +14,8 @@ namespace Microsoft.VisualStudio.Packaging
 {
     [Guid(PackageGuid)]
     [PackageRegistration(AllowsBackgroundLoading = true, RegisterUsing = RegistrationMethod.CodeBase, UseManagedResourcesOnly = true)]
+    [RemoteCodeGeneratorRegistration(ResXGenerators.ResXGuid, ResXGenerators.ResXGeneratorName, ResXGenerators.ResXGeneratorName, @"{" + ProjectTypeGuid + @"}", GeneratesDesignTimeSource = true)]
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/Packaging/CSharpProjectSystemPackage.cs]: Can you add all the other single file generators that old csproj registers as well? Especially the texttemplate one keeps coming up.

**Comment by 333fred** [comment]
Tagging @dotnet/project-system for review.

File: src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/Packaging/CSharpProjectSystemPackage.cs (hunk 1 above)
Comment (333fred) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/Packaging/CSharpProjectSystemPackage.cs]: Do all of the ones that are currently registered make sense? For example, .XAML generators? I've put the full list in #218.

Comment (333fred) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/Packaging/CSharpProjectSystemPackage.cs]: Also, I'll add an attribute that allows us to register file extensions with a generator, so we get .tt support.

**Comment by 333fred** [comment]
@dotnet-bot retest this please

**Comment by 333fred** [comment]
@dotnet-bot retest release please

File: src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/Packaging/CSharpProjectSystemPackage.cs [hunk 2]
```diff
@@ -13,6 +14,16 @@ namespace Microsoft.VisualStudio.Packaging
 {
     [Guid(PackageGuid)]
     [PackageRegistration(AllowsBackgroundLoading = true, RegisterUsing = RegistrationMethod.CodeBase, UseManagedResourcesOnly = true)]
+    [RemoteCodeGeneratorRegistration(SingleFileGenerators.ResXGuid, SingleFileGenerators.ResXGeneratorName,
+        SingleFileGenerators.ResXDescription, "{" + ProjectTypeGuid + "}", GeneratesDesignTimeSource = true)]
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/Packaging/CSharpProjectSystemPackage.cs]: Maybe have another constant that includes the curlies for brevity?

**Comment by srivatsn** [comment]
👍

**Comment by 333fred** [comment]
@Srivatsn, do I need to include anything for VB in here as well?

**Comment by srivatsn** [comment]
Ah yes please do.

**Comment by 333fred** [comment]
I've retried the release build twice, still no luck on getting past that vsix bug.

**Pull request merged**